File: algo/lc.array.21.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0: return 0
        idx = 1
        for i in range(1, len(nums)):
            if nums[i] > nums[idx-1]:
                nums[idx] = nums[i]
                idx += 1
        return idx

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums: return 0
        idx = 0
        for j in nums:
            if j != nums[idx]:
                idx += 1
                nums[idx] = j
        return idx+1

File: algo/lc.array.21.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut idx: usize = 0;
        for j in &mut nums {
            if j != nums[idx] {
                idx += 1;
                nums[idx] = j;
            }
        }
        return (idx + 1) as i32;
    }
}
